cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -393,7 +398,8 @@ public void run() {
           }
           alterExploreStream(StreamUtils.getStreamIdFromLocation(streamLocation), false);
           configLocation.delete();
-        } catch (IOException e) {
+          streamMetaStore.removeStream(streamId);
+        } catch (Exception e) {
```

Comment:
it does a little bit in that any run time exception that would have happened before will now get wrapped again because of Throwables.propagate